proposals/11385-literal-input-parameters/README.md

```diff
@@ -0,0 +1,138 @@
+will validate the input values against their Literal parameter values, if applicable 
+(For example, if a hard-coded pipeline-level input value is passed into a
+Literal parameter at the component level, the SDK will validate the input). If input is invalid, compilation fails.
+#### API Server
+The following changes to the API server should be implemented next because the pipeline run server changes require a compiled pipeline YAML file.
+- Extend the pipeline run server runtime parameter validation logic to check the “literals” field of a ComponentInputsSpec_ParameterSpec. If “literals” is non-empty, then the runtime parameter should be checked against the valid input options. This logic lives in `v2_template.validatePipelineJobInputs()`.
+#### Driver
+The following changes to the Driver should be implemented next:
+- Add a switch case to `backend/src/v2/driver.resolveInputParameter()` for processing and validating Literal parameters. If a pipeline passes the output of one component as the input for a Literal parameter into a second component,
+  this value must be validated.
+#### UI
+The update to the UI should be implemented last, because it depends on the SDK and API server changes.
+- When a pipeline YAML file containing a pipeline with Literal input is uploaded via the UI, the box for each input parameter will display a drop-down list containing the one or more values contained within the Literal parameter for the user to click. The user should not be able to type an input in the box - options should be selection-only.
+### Test Plan
+- Each of the following test cases can be implemented with string, int and float Literal parameters. 
+- Hard-coded input refers to input passed into the pipeline function call. This is in contrast to runtime input, which is passed in when the compiled pipeline is submitted to the API server.
+#### The following test cases are expected to pass: 
+|                                                           | SDK Compiler | Workflow Compiler | Execute Pipeline E2E | API Server: Verify Pipeline Run |
+|-----------------------------------------------------------|--------------|-------------------|----------------------|----------------------------|
+| Pipeline-level Literal input:<br/> valid hard-coded input | ✓            | ✓                 | ✓                    | ✓                          |
+| Pipeline-level Literal input:<br/>no input (SDK only)     | ✓            | X                 | X                    | ✓                          |
+| Component-level Literal input                             | ✓            | ✓                 | ✓                    | ✓                          |
+| Pipeline & component-level:<br/> valid hard-coded input   | ✓            | ✓                 | ✓                    | ✓                          |
+| Pipeline-level Literal input:<br/> valid runtime input    | X            | ✓                 | ✓                    | ✓                          |
```

Comment:
Also include the case where the output of a component is the input to a literal input parameter.